tests/kernel/threads/lifecycle/lifecycle_api/src/test_essential_thread.c

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2016 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <ztest.h>
+#include <kernel.h>
+#include <kernel_structs.h>
+
+__kernel struct k_thread kthread_thread;
+
+#define STACKSIZE 1024
+#define PIPE_SIZE 16
+#define BYTES_TO_WRITE 4
+#define BYTES_TO_READ BYTES_TO_WRITE
+
+static unsigned char __aligned(4) data[] = "ATestOnEssential";
+
+K_PIPE_DEFINE(my_pipe, PIPE_SIZE, 4);
+K_THREAD_STACK_DEFINE(kthread_stack, STACKSIZE);
+K_SEM_DEFINE(sync_sema, 0, 1);
+
+static void pipe_get(struct k_pipe *my_pipe)
+{
+	unsigned char rx_data[PIPE_SIZE];
+	size_t to_read, read_byte = 0;
+
+	for (int i = 0; i < PIPE_SIZE; i += read_byte) {
+		to_read = (PIPE_SIZE - i) >= BYTES_TO_READ ?
+			BYTES_TO_READ : (PIPE_SIZE - i);
+		zassert_false(k_pipe_get(my_pipe, &rx_data[i], to_read,
+					&read_byte, 1, K_FOREVER), NULL);
+		zassert_true(read_byte == to_read || read_byte == 1, NULL);
+	}
+	for (int i = 0; i < PIPE_SIZE; i++) {
+		zassert_equal(rx_data[i], data[i], NULL);
+	}
+}
+
+static void thread_entry(void *p1, void *p2, void *p3)
+{
+	_thread_essential_set();
+	if (_is_thread_essential()) {
+		pipe_get((struct k_pipe *)p1);
+	} else {
+		zassert_unreachable("The thread is not set as essential\n");
+	}
+	_thread_essential_clear();
+	k_sem_give(&sync_sema);
+}
+
+static void pipe_put(struct k_pipe *my_pipe)
+{
+	size_t to_write, write_byte = 0;
+
+	for (int i = 0; i < PIPE_SIZE; i += write_byte) {
+		to_write = (PIPE_SIZE - i) >= BYTES_TO_WRITE ?
```

Comment:
this logic looks complicated, what do we want to do here?

if we want to send 1 byte at a time , just send one byte else you can send all 4 bytes.

same applies fro pipe_read

Reply:
@ramakrishnapallala: The data size is 16 bytes, the loop basically writes 4 bytes at every iteration and checks if all 4 bytes are written. Minimum number of bytes it can transfer is 1. There is a check for write_byte. There is a zassert check where write_byte value has to be 4 or more than 1 for every iteration. I can make minimum number of bytes as 4 itself, so that it will be simpler.